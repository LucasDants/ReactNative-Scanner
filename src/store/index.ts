import { IQRCodeList } from './modules/qrlist/types'
import rootReducer from './modules/rootReducer'
import { configureStore } from '@reduxjs/toolkit'

export interface IState {
    qrlist: IQRCodeList
}

const store = configureStore({reducer: rootReducer}) 

export default store