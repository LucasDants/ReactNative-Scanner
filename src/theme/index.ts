export default {
  colors: {
    red: {
      900: '#B83341',
      800: '#E03F50',
    },
    blue: '#0000FF',
    white: '#FFFFFF',
    gray: {
      100: '#DCDCDC',
      600: '#404040',
      900: '#030303',
    },
  },
};