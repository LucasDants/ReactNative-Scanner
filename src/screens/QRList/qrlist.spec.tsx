import React from 'react'

import { fireEvent, render } from '@testing-library/react-native'

import { ThemeProvider } from 'styled-components/native'
import theme from '../../theme'

import { QRList } from '.'
import { Provider } from 'react-redux'
import store from '../../store'

const Providers = ({ children }) => (
  <Provider store={store}>
    <ThemeProvider theme={theme}>
      { children }
    </ThemeProvider>
  </Provider>
)

const qrlistData = ['code', 'mode', 'testing']
const dispatchActions = qrlistData.map(QRCode => ({type: 'ADD_QRCODE_TO_LIST', payload: { QRCode }})) 

describe('QRList Screen', () => {

  it("should match the snapshot test", () => {
    const { toJSON } = render(
      <QRList />, {wrapper: Providers}
    )

    const component = toJSON()
    expect(component).toMatchSnapshot()

  })

  it("should have an text warning when the list is empty", () => {
    const { getByText } = render(
      <QRList />, {wrapper: Providers}
    )

    expect(getByText("You don't have QRCodes yet! Read some to appear in the list!")).toBeTruthy()
  })

  it('should have a list of qrcodes', () => {
    store.dispatch(dispatchActions[0])
    store.dispatch(dispatchActions[1])
    store.dispatch(dispatchActions[2])
    
    const { getByText } = render(
      <QRList />, {wrapper: Providers}
    )

    expect(getByText("code")).toBeTruthy()
    expect(getByText("mode")).toBeTruthy()
    expect(getByText("testing")).toBeTruthy()
  })

  it('should filter the list of qrcodes by the search input value', () => {
    const { getByText, getByTestId, queryByText } = render(
      <QRList />, {wrapper: Providers}
    )

    const input = getByTestId('input')
    const buttonSearch = getByTestId('button-search')

    fireEvent.changeText(input, 'ode')
    fireEvent.press(buttonSearch)
    
    expect(getByText("code")).toBeTruthy()
    expect(getByText("mode")).toBeTruthy()
    expect(queryByText("testing")).toBeFalsy()
  })

  it('should clear the search input and the show the original list', () => {
    const { getByText, getByTestId, queryByText } = render(
      <QRList />, {wrapper: Providers}
    )

    const input = getByTestId('input')
    const buttonSearch = getByTestId('button-search')
    const buttonClear = getByTestId('button-clear')

    fireEvent.changeText(input, 'ode')
    fireEvent.press(buttonSearch)

    expect(getByText("code")).toBeTruthy()
    expect(getByText("mode")).toBeTruthy()
    expect(queryByText("testing")).toBeFalsy()
    
    fireEvent.press(buttonClear)

    expect(input.props.value).toBe('')

    expect(getByText("code")).toBeTruthy()
    expect(getByText("mode")).toBeTruthy()
    expect(getByText("testing")).toBeTruthy()

  })

})