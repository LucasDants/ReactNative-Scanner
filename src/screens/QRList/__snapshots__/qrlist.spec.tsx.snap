// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QRList Screen should match the snapshot test 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#030303",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <ViewManagerAdapter_ExpoLinearGradient
    proxiedProperties={
      Object {
        "colors": Array [
          4290261825,
          4292886352,
        ],
        "endPoint": undefined,
        "locations": undefined,
        "startPoint": undefined,
      }
    }
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderBottomColor": "#B83341",
          "borderBottomWidth": 1,
          "justifyContent": "center",
          "paddingBottom": 10,
          "paddingLeft": 24,
          "paddingRight": 24,
          "paddingTop": 33,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 51,
          },
        ]
      }
    >
      QRCode List
    </Text>
  </ViewManagerAdapter_ExpoLinearGradient>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "paddingBottom": 10,
          "paddingLeft": 12,
          "paddingRight": 12,
          "paddingTop": 10,
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#404040",
            "borderRadius": 6,
            "flexBasis": 0,
            "flexDirection": "row",
            "flexGrow": 1,
            "flexShrink": 1,
          },
        ]
      }
    >
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Search QRCodes..."
        placeholderTextColor="#DCDCDC"
        rejectResponderTermination={true}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "height": 52,
              "paddingLeft": 12,
            },
          ]
        }
        testID="input"
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "marginRight": 7,
            "opacity": 1,
            "paddingBottom": 12,
            "paddingLeft": 12,
            "paddingRight": 12,
            "paddingTop": 12,
          }
        }
        testID="button-clear"
      >
        <Text />
      </View>
    </View>
    <RNGestureHandlerButton
      activeOpacity={0.105}
      collapsable={false}
      onActiveStateChange={[Function]}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#B83341",
          "borderRadius": 6,
          "height": 52,
          "justifyContent": "center",
          "marginLeft": 7,
          "width": 52,
        }
      }
      testID="button-search"
      underlayColor="black"
    >
      <View
        style={
          Object {
            "backgroundColor": "black",
            "borderBottomLeftRadius": undefined,
            "borderBottomRightRadius": undefined,
            "borderRadius": 6,
            "borderTopLeftRadius": undefined,
            "borderTopRightRadius": undefined,
            "bottom": 0,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
      <Text />
    </RNGestureHandlerButton>
  </View>
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    ListEmptyComponent={[Function]}
    contentContainerStyle={
      Object {
        "padding": 12,
      }
    }
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    style={
      Array [
        Object {},
      ]
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 31,
              "lineHeight": 20,
              "textAlign": "center",
            },
          ]
        }
      >
        You don't have QRCodes yet! Read some to appear in the list!
      </Text>
    </View>
  </RCTScrollView>
</View>
`;
